### ISSUE TITLE
RCS1097 - context on when something is/isn't a false negative

### ISSUE THREAD

**Issue by mgravell** [JosefPihrt/Roslynator#422]
Related to #87

The following example (based on the example from the above) triggers RCS1097; the previously logged issue was specifically to *make* this log. However, IMO (and this may be subjective, I realize) this is **not** redundant. My reasoning for saying this is because it [results in a different overload being invoked](https://sharplab.io/#v2:C4LglgNgPgrgzmAdgcwAQGUCedgFMC2A3ALABQZiAhvrnAA6UDGuqAYjIgOpjAAWASgGF0ARgAMATgDsZAN5lUi1EjwAnKhFQ5KwMI1SMIlOHFQAFVLNR1VYAG46W23frsB7MABNUAWUpIACgBKS1QAX3CyBSU6GAAjCD0DIxNUQWjFeVIlHOt4xNcPbx9gjNys3MqtYFsUVAAvVABeVERcAHdUAHk4gCtcRmBggDoAFTd0GqRkYNQAalQAIjwcRfnUACl4YHWfN1UWBcmYADMTkmyqsKjLypt7R1QDyk83RAhMatq0LZwAGl8+1wAOOZwuZRysQSSXcXlQADlSrdyhCqjhvg1mq0Ot0+gMhiEFstaMA1gtfjsFnsDutQedUYprsimUyyEA=). The *final outcome* ends up the same, but the `object[]` version allocates an extra `string[]` per call, [here](https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/String.Manipulation.cs#L102). It then calls `ToString()` per item internally, but: it is a different overload call with a different behaviour. IMO, therefore, it is not "redundant" - keeping the `ToString()` actually prevents an allocation.

If anything, I want people to *add* `ToString()` calls in this scenario, not remove them :)

Thoughts?

```c#
using System;

namespace FunWithRCS1097
{
    internal static class P { private static void Main() { } }

    public class C
    {
        public void M()
        {
            string z = new Object().ToString() + "test" + Just + More + Stuff;
        }

        private readonly string Just, More, Stuff;

        public void N()
        {
            string z = new Object() + "test" + Just + More + Stuff;
        }
    }
}
```
